@AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/fcr/fn_fcrLongbow.sqf

```diff
@@ -10,42 +10,65 @@ Returns:
 Examples:
     [_heli] spawn fza_fnc_fcrLongbow;
 Author:
-    Rosd6(Dryden), BradMick
+    Rosd6(Dryden)
 ---------------------------------------------------------------------------- */
 if (!(isNil "fza_ah64_nofcr")) exitwith {};
 _heli = vehicle player;
+if !((vehicle player) animationphase "plt_apu" > 0.5 || (isEngineOn _heli)) exitwith {};
 
 #define AGMODE_GND 0
 #define AGMODE_AIR 1
-_dataLinkArray = [];
+_datalinkarray = [];
 
-if ((vehicle player) animationphase "plt_apu" > 0.5 || (isEngineOn _heli)) then {
-	if (isVehicleRadarOn _heli && (_heli animationPhase "fcr_enable" == 1) && _heli getHit "radar" < 0.8) then {
-        _dataLinkArray = (listRemoteTargets playerSide select {_x # 1 > 6}) apply {_x # 0};
-		{
-			if (alive _x) then {
-				_distOffAxis = abs ([[_heli, (getposatl _heli select 0), (getposatl _heli select 1), (getposatl _x select 0), (getposatl _x select 1)] call fza_fnc_relativeDirection] call CBA_fnc_simplifyAngle180);
+if (isVehicleRadarOn _heli && (_heli animationPhase "fcr_enable" == 1) && _heli getHit "radar" < 0.8) then {
+	_datalinkarray = (listRemoteTargets playerSide) apply {_x # 0};
+	{
+		_distOffAxis = abs ([[_heli, (getposatl _heli select 0), (getposatl _heli select 1), (getposatl _x select 0), (getposatl _x select 1)] call fza_fnc_relativeDirection] call CBA_fnc_simplifyAngle180);
+		if (_x == _heli || _x isKindOf "man" || _x isKindOf "StaticCannon" || laserTarget _heli == _x) then {
+			_dataLinkArray = _dataLinkArray - [_x];
+		};
+		if !(alive _x) then {
+			_dataLinkArray = _dataLinkArray - [_x];
```

Comment:
Even if the previous condition removed the item from the data link array, this code still goes through and continues doing checks, and maybe tries to remove it again from the array (as mentioned before this is an expensive operation.)

Maybe you could take advantage a brand new operator, [`continue`](https://community.bistudio.com/wiki/continue) to skip the rest of the checks if they aren't required?

Reply:
seems like it will work

Reply:
while this does work, in testing it cause more fps drops with it vs without it
40fps with v 58 fps without